### PR TITLE
Refactor avatar to use a common client and proxy, add Azure Live Avatar

### PR DESCRIPTION
**Pull request by juberti**
This defines a common client (RestPeerConnectionClient) that factors out the individual provider signaling into derived implementations. A signaling proxy is used on the server side to allow the individual clients to speak each individual signaling protocol, but where the proxy adds the API key before forwarding to the right provider (which avoids the need for individual endpoints on the server for each provider).

The Azure Live Avatar speaks its own custom protocol via a websocket, but the details of that are hidden away in a SDK so we can mostly just map our operations to the SDK, and then we can still use the PeerConnectionClient base class to get consistent behavior in the rest of the code.

Next steps:
- Yepic impl
- finish transparency, do a transform or shader to convert the green pixels to transparent ones
- hook up some basic latency stats

### CODE REVIEW

File: src/app/avatar/page.tsx [hunk 1]
```diff
@@ -1,78 +1,140 @@
...
+    }
+  };
+  const generate = async (provider: Provider) => {
+    if (!clientRef.current || provider != currentProvider) {
+      clientRef.current?.close();
+      await connect(provider);
+    }
+    await clientRef.current!.generate({ text: { text } });
+  };
+
```
Comment (liPatrick) [inline src/app/avatar/page.tsx:110]: Should we add a method to teardown the client connection once we switch to a different provider?

**Review by liPatrick — APPROVED** [review]
Thanks for this, much cleaner now!

File: src/app/avatar/page.tsx (hunk 1 above)
Comment (juberti) [reply src/app/avatar/page.tsx:110]: I think that's on line 105, right?

Comment (liPatrick) [reply src/app/avatar/page.tsx:110]: Yes! I missed that sorry

**Pull request merged**